quinn-proto/src/connection.rs

```diff
@@ -3166,9 +3174,12 @@ where
     K: crypto::Keys,
 {
     crypto: K,
-    end_packet: u64,
-    /// Time at which a packet using the following key phase was received
-    update_ack_time: Option<Instant>,
+    /// The first incoming packet to use the next key phase, and the time at which it was received
+    ///
+    /// Incoming packets with a lower number should be decrypted with `crypto`. `None` indicates
+    /// that we have not yet received a packet using newer keys, which is only possible if the
+    /// update was locally initiated.
+    end_packet: Option<(u64, Instant)>,
```

Comment:
I find the naming of `end_packet` combined with the description `The first incoming packet (...)` quite confusing. Maybe it should be named `first_packet`?

Reply:
Reworded the comment to hopefully be clearer. The current name is correct, because it represents the end of the incoming packet number interval for which the previous keys (which are stored in this structure) should be used.

Reply:
Hmm, it's still confusing to me. `The incoming packet that triggered or acknowledges [sic] (...)` still seems to be about a first packet of the next range. Maybe the difference is in referencing right-inclusive or right-exclusive ranges? In which case I would suggest this ought to be named `last_packet`.

Reply:
> still seems to be about a first packet of the next range

Yes, that's what the exclusive end-point of the previous range is.

> last_packet

The packet in question is part of the next range, not the previous one.

Reply:
Made another attempt at rephrasing.